### PR TITLE
feat:  improved metrics and added rate limit for the batch request feature in ws server (#2474)

### LINKED ISSUE #2474: Improve metrics and add rate limit for the batch request feature in WS server
### Problem

Currently, batch request feature lack of metrics and is not rate limited.

### Solution

Improve metrics and add rate limit for the batch request feature in WS server

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by quiet-node**
**Description**:
- added `shouldRateLimitOnMethod()` function to ConnectionLimiter class which utilizes the `shouldRateLimit()` method from `RateLimit` class to apply rate limit logic on both batch requests and single requests in the WS server
- added methodsCounter metric for `batch_requests` methods
- added a list of acceptancetest for rateLimit in the WS server
- modified `acceptancetest:ratelimiter` script to run tests with `@web-socket-ratelimiter` tag. This change will add the new acceptancetest to the `ratelimiter` CI workflow

**Metrics result**:
**Startup:
image

** After multiple batch requests calls:
image

Fixes #2474

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/ws-server/src/webSocketServer.ts [hunk 1]
```diff
@@ -135,7 +145,13 @@ app.ws.use(async (ctx) => {
         return;
       }
 
-      // TODO: verify rate limit
+      // verify rate limit for batch_request method based on IP
+      if (limiter.shouldRateLimitOnMethod(ctx.ip, WS_CONSTANTS.BATCH_REQUEST_METHOD_NAME, ctx.websocket.requestId)) {
```
Comment (Nana-EC) [inline packages/ws-server/src/webSocketServer.ts:149]: Tests?
I think the relay has a test setup that you can mimic e.g. set the rate limit small and send requests to exceed it and verify

**Review by Nana-EC — CHANGES_REQUESTED** [review]
LG.
Missing acceptance tests

**Review by AlfredoG87 — CHANGES_REQUESTED** [review]
as @Nana-EC pointed is missing UT for rate limits.

also, atm we don't do automated tests for `metrics` but we usually add a screenshot of the `/metrics` endpoint where you can see the metrics added with some values on **Notes for reviewer:** section of the PR description.

**Comment by quiet-node** [comment]
> as @Nana-EC pointed is missing UT for rate limits.
> 
> also, atm we don't do automated tests for `metrics` but we usually add a screenshot of the `/metrics` endpoint where you can see the metrics added with some values on **Notes for reviewer:** section of the PR description.

added prrof for metrics

**Review by AlfredoG87 — APPROVED** [review]
LGTM

**Review by Nana-EC — APPROVED** [review]

File: .github/workflows/acceptance.yml [hunk 2]
```diff
@@ -47,6 +47,7 @@ jobs:
     uses: ./.github/workflows/acceptance-workflow.yml
     with:
       testfilter: ratelimiter
+      test_ws_server: true
```
[the remark this answers is not in the document]
Comment (quiet-node) [reply .github/workflows/acceptance.yml:50]: > Q: what's this config and why is it just for ws.

This config sets the `process.env.TEST_WS_SERVER` variable in the [CI job](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/.github/workflows/acceptance-workflow.yml#L87). The test program uses this `TEST_WS_SERVER` environment variable to determine whether to run the WS server, as shown [here](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/ws-server/tests/acceptance/index.spec.ts#L175).

> Shouldn't we test all the time or if there is a config shouldn't there be on for the HTTP and the WS server each?
We didn't have tests for rate limiting on the WS server until this PR. Previously, we only had them for the HTTP server. That's why this workflow has been running exclusively against the HTTP server.

**Pull request merged**